examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -829,6 +829,49 @@ fn contract_serialization_helper<S: HasStateApi>(
     Ok(())
 }
 
+/// Helper function to calculate the `message_hash`.
+#[receive(
+    contract = "cis3_nft",
+    name = "viewMessageHash",
+    parameter = "PermitParam",
+    return_value = "[u8;32]",
+    crypto_primitives,
+    mutable
+)]
+fn contract_view_message_hash<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &mut impl HasHost<State<S>, StateApiType = S>,
+    crypto_primitives: &impl HasCryptoPrimitives,
+) -> ContractResult<[u8; 32]> {
+    // Parse the parameter.
+    let mut cursor = ctx.parameter_cursor();
+    // The input parameter is `PermitParam` but we only read the initial part of it
+    // with `PermitParamPartial`. I.e. we only read the `signature`, the
+    // `signer` but WITHOUT the `message` here.
+    let param: PermitParamPartial = cursor.get()?;
+
+    // The input parameter is `PermitParam` but we only read the initial part of it
+    // with `PermitParamPartial` so far. We read in the `message` now.
+    // `(cursor.size() - cursor.cursor_position()` is the length of the message in
+    // bytes.
+    let mut message_bytes = vec![0; (cursor.size() - cursor.cursor_position()) as usize];
+
+    cursor.read_exact(&mut message_bytes)?;
+
+    // The message signed in the Concordium browser wallet is prepended with the
+    // `account` address and 8 zero bytes.
```

Comment:
What are the eight zero bytes? Could you explain it in the comment? 😊

Reply:
The 8 bytes zero are just a replication of what the browser wallet is doing at the moment. The 8 zero bytes could be used in the future in case we want to add versioning or we want to distinguish different signing mechanisms but for the time being there is no further meaning of the 8 zero bytes. As long as we don't have a good explanation for what the 8 zero bytes could represent in the future, I would leave the sentence as it is.

Reply:
The 8 zero bytes are quite important to be all 0.

This makes sure the user is not accidentally signing a transaction!

Reply:
Could you elaborate on what is encoded instead of the 8 zero bytes in a normal transaction (which will never be 0) then I can add a sentence about how these 8 bytes are used to distinguish between signing a message and signing a transaction, please.